app/models/user.rb

```diff
@@ -5,6 +5,16 @@ class User < ApplicationRecord
 
   has_many :microposts, dependent: :destroy
 
+  has_many :active_relationships, class_name: Relationship.name,
```

Comment:
giải thích cho a các option này, và lý do tại sao cần dùng thêm các option này nhé class_name, foreign_key, through, source

Reply:
through và source : biểu hiện quan hệ n-n thông qua (through) bảng trung gian ? User gọi following/followers thông qua source followed/follower trong relationship
class_name: cho Rails biết thằng đang gọi nó thuộc về model nào.
foreign_key: Khóa ngoại, chức năng như trên nhưng cho biết gán theo key nào ?

Reply:
Nguồn tham khảo :|
https://viblo.asia/p/option-in-association-source-as-and-source-type-nhung-dieu-thu-vi-ve-association-co-the-ban-da-biet-oOVlYy4vl8W

Reply:
@PhongPhamdt `và lý do tại sao cần dùng`

Reply:
Theo logic ban đầu, mn sẽ nghĩ là cần tạo bảng riêng là following chứa người follow và người được follow. Nhưng như vậy sẽ gây ra sự "redundant" vì id các user sẽ lặp đi lặp lại tạo thành n^n bản ghi.
Ta gộp cả followed và follower trong bảng Relationship nên cần define thành active_relationship và passive_relationship 
User có 2 khóa ngoại đến followed_id và follower_id trong bảng Relationship, truy cập đến các followed hay follower dùng through và source

Reply:
@PhongPhamdt đọc tutorial đi, nó có đoạn loạnh quanh chỗ mấy cái hình vẽ nói ly do cần dùng các option trên đó
anh đang hỏi lý do cần dùng các option tren, chứ k phải lý do thiết kế n-n

Reply:
foreign_key/class_name: khóa ngoại của bảng User đến bảng <blah blah> được rails mặc định là <blah blah>_id, trong Model này, ta cần cả 2 khóa ngoại từ bảng User nên cần define bằng foreign_key.
Vì tạo n-n như trên, cần dùng through và source để gọi following từ followed_id, followers từ follower_id trong relationship